packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyAction.tsx

```diff
@@ -512,6 +525,33 @@ export const GroupPropertyAction = ({
           </Button>
         </ModalButtonBar>
       </Modal>
+      <Modal
+        title='Confirm'
+        modalRootId='grouping-mapping-widget'
+        isOpen={showSaveModal}
+        onClose={() => {
+          setShowSaveModal(false);
+        }}
+      >
+        <div className="gmw-save-modal-body-text">
+          <Text variant="leading" as="h3">
+            Are you sure you want to save this property with a new name? You may need to update this name in Custom Calculation formulas.
+          </Text>
+        </div>
+        <ModalButtonBar>
+          <Button styleType='high-visibility' onClick={onSave}>
+            Save
+          </Button>
+          <Button
+            onClick={() => {
+              setShowSaveModal(false);
+            }}
+            styleType='default'
+          >
+            Cancel
+          </Button>
+        </ModalButtonBar>
+      </Modal>
```

Comment:
This looks like it can be made into its own component with its own file.

Reply:
Made into a separate component.